### DISCUSSION TITLE
update pdf info API call from 0.2.5 to 0.5.0

### DISCUSSION THREAD

**Discussion by sabrinajestar** [pdfcpu/pdfcpu#718] [Q&A]
Hello,

My existing code, which uses api version 0.2.5, looks like this:

`metadata, err := pdfcpuapi.InfoFile(path, pdfcpu.NewDefaultConfiguration())`

I just want to draw the metadata from a PDF into a map. Nothing fancy (no need for specifying selected pages, no custom configuration options).

I am trying to update this code for 0.5.0, but am struggling to work this out. There is no obvious replacement for InfoFile() that accepts a similar list of params. Info() and ListInfo() and PDFInfo() require an entirely different set of parameters. There also seems to be no replacement for NewDefaultConfiguration(). Can you suggest what code would take the place of the line above? Thanks!

**Comment by hhrutter** [comment]
Hi there,

**api.PDFInfo** was refactored and now returns a struct with all the info you need.

You can either use that or if you prefer a formatted string slice then you'd probably go for: **cli.ListInfoFile**

See also pkg/cli/list.go: func ListInfoFiles as far as usage
